app/_components/DistanceHintButton.tsx

```diff
@@ -30,7 +30,7 @@ function DistanceHintButton({
       handleDistanceHintButtonState(position.coords);
     });
     return () => navigator.geolocation.clearWatch(id);
-  }, []);
+  });
```

Comment:
Removing the dependency array here means that this `useEffect` will run on every render, which could lead to undesired/uncontrolled behavior.